config/nrfconnect/chip-module/CMakeLists.txt

```diff
@@ -106,6 +106,14 @@ endif()
 
 get_property(CHIP_COMPILER_LAUNCHER GLOBAL PROPERTY RULE_LAUNCH_COMPILE)
 
+# Usage of env is because `west` seems to use a hierarcical build system
```

Comment:
note: you can pass CMake variables using west in the following way: `west build ... -- -DSOME_CMAKE_VAR=1` (all arguments following `--` are passed to CMake).

Reply:
I tried that and it did not work (although if you have a way to make it work that would be 100% better.

What I saw is that top level and mcuboot subdirectories had it in their CMakeCache.txt however the nrfconnect subdirectory had its own CMakeCache.txt hence my comment here and ugly workaround. 

I thought it is related to sysbuild which says I should namespace it: https://docs.zephyrproject.org/latest/build/sysbuild/index.html#cmake-variable-namespacing  however I tried that and it did not work for me at that point either.

Reply:
Shouldn't we use a Kconfig to control this feature ? This is what is usually done for all zephyr based platforms.

Reply:
I need to control build logic: included source files and flags passed to GN. It did not seem like Kconfig flags are used for that (or at least I did not seem to have them visible inside cmakelists when I tried this).

Reply:
also as an aside, I did try for nrf as a namespace all of `nrf/nrfconnect/matter/chip/zephyr` and none transferred fom `-D....=...` in the cmake command line into nrfconnect/CMakeCache (which is where I need them). As a result, the existing logic with the apology comment is the only thing that I could make work.

Hoping there is a better solution however I cannot find it and also longer term I hope we can fully transition things and eventually remove it: what I have today is to allow for incremental development and testing.

Reply:
What I meant is that usually, we map GN args with a Kconfig entry.
As an example, you could add a new Kconfig entry in `config/zephyr/Kconfig`, something like this:
```
config CHIP_DATA_MODEL_INTERFACE
    bool "Use data model interface"
    help
      something
```

then, in zephyr based platforms CMakeLists.txt, like `nxp/chip-module/CMakeLists.txt`, you can pass the argument to gn like this:
```
matter_common_gn_args(
    DEBUG           CONFIG_DEBUG
    LIB_SHELL       CONFIG_CHIP_LIB_SHELL
    LIB_TESTS       CONFIG_CHIP_BUILD_TESTS
    PROJECT_CONFIG  ${CONFIG_CHIP_PROJECT_CONFIG}
    DEVICE_INFO_EXAMPLE_PROVIDER CONFIG_CHIP_EXAMPLE_DEVICE_INFO_PROVIDER
    DATA_MODEL_INTERFACE CONFIG_CHIP_DATA_MODEL_INTERFACE
)
```
You might have to convert the boolean to your "enabled/disabled" string in `matter_common_gn_args`

I think this is preferred over using environment variables, as we can then configure the Kconfig value at application level (in the prj.conf files).

Reply:
kconfig value setting did carry over. Updated PR. Thank you!